Add blog post for 4.4 dev1 (#911)
godotengine/godot-website: collections/_article/dev-snapshot-godot-4-4-dev-1.md

gathered a lot of changes that warrant a dev release!

Most of the changes in this release are bug fixes that will be backported to Godot 4.3 and released in 4.3.1! So please
test this release well so we can be confident with the changes and release 4.3.1 with them as soon as possible.

Keep in mind that while we try to make sure each dev snapshot is stable enough for general testing, this is by
definition a pre-release piece of software. Be sure to make frequent backups, or use a version control system such as
Git, to preserve your projects in case of corruption or data loss.

[Jump to the **Downloads** section](#downloads), and give it a spin right now, or continue reading to learn more about improvements in this release. You can also [try the **Web editor**](https://editor.godotengine.org/releases/4.4.dev1/) or the **Android editor** for this release. If you are interested in the latter, please request to join [our testing group](https://groups.google.com/g/godot-testers) to get access to pre-release builds.

-----

*The cover illustration is from* [**TODO**](TODO), * by [TODO](https://TODO).*

## Highlights

> After only a week of development we are bringing you a bunch of fixes and improvements

Reviewer: ```suggestion
After only a week of development, we are bringing you a bunch of fixes and improvements:
```

### Metal Rendering Backend

We hinted that a Metal backend was in the works and it is finally here ([GH-88199](https://github.com/godotengine/godot/pull/88199))! 

> Metal is a low-level graphics API similar to Vulkan or D3D12. Currently Godot supports using Vulkan or D3D12, but
> neither of those are available on MacOS and iOS, so we use a library called [MoltenVK](https://github.com/KhronosGroup/MoltenVK)
> to run Vulkan over Metal. MoltenVK is great and has worked well for us. But having our own Metal implementation is more efficient
> and allows us to have a greater control over what features we support and what performance tradeoffs we make. 

Reviewer: ```suggestion
Metal is a low-level graphics API similar to Vulkan or D3D12. Godot currently supports using Vulkan or D3D12, but
neither of those are available on MacOS and iOS, so we use a library called [MoltenVK](https://github.com/KhronosGroup/MoltenVK)
to run Vulkan over Metal. MoltenVK is great and has worked well for us. However, having our own Metal implementation is more efficient
and allows us to have a greater control over what features we support and what performance tradeoffs we make. 
```

Earlier results show that the Metal backend on MacOS is at least as fast as the Vulkan backend and in many cases, much faster.

> Right now we only support Metal on Apple silicon devices. That includes all iOS devices and the M1/M2/M3 Macs. Other
> Apple devices will continue to work with the MoltenVK backend. Currently this limitation is simply due to the fact that
> few contributors working in this area have access to non-Apple Silicon devices. 

Reviewer: ```suggestion
Right now, we only support Metal on Apple Silicon (ARM) devices. That includes all iOS devices and the M1/M2/M3 Macs. Intel-based
Apple devices will continue to work with the MoltenVK backend. Currently, this limitation is due to the fact that
few contributors working in this area have access to non-Apple Silicon devices. 
```

### 3D physics interpolation
Godot 4.3 came with physics interpolation for 2D nodes, and now the 3D counterpart has been merged ([GH-92391](https://github.com/godotengine/godot/pull/92391))!
Not only that but support for MultiMeshes has also been merged ([GH-91818](https://github.com/godotengine/godot/pull/91818))!

Physics interpolation is a technique that allows you to run your physics update at a very low FPS while maintaining
smooth movement. This allows you to both save CPU overhead and make your game look much smoother. 

<video autoplay loop muted playsinline>
  <source src="/storage/releases/4.4-dev1/physics-interpolation.mp4?1" type="video/mp4">
</video>

### Lightmap Bicubic sampling
Bicubic sampling is a method for reading from a lightmap that smooths our sharp edges. It is especially useful when
you bake lightmaps with shadows at a low texture resolution. We supported this in Godot 3, and have now brought it back in
[GH-89919](https://github.com/godotengine/godot/pull/89919).

![Comparison between shadows with bicubic on and off](/storage/releases/4.4-dev1/bicubic-compare.webp)

> Bicubic sampling does come with a small run-time performance cost. So it can be disabled in a project setting if needed.

Reviewer: ```suggestion
Bicubic sampling comes with a small run-time performance cost on the GPU, so it can be disabled in a project setting if needed.
```

- [Core] Optimize Array.insert ([GH-94353](https://github.com/godotengine/godot/pull/94353)). This makes inserting new
    elements into an Array about 10X faster on average. 
- [Core] Optimize `String.insert` ([GH-92555](https://github.com/godotengine/godot/pull/92555)).
- [Core] Optimize `String.join` ([GH-92550](https://github.com/godotengine/godot/pull/92550)).
- [Core] Optimize `String.replace` ([GH-92546](https://github.com/godotengine/godot/pull/92546)).
- [Import] Disable normal raycaster by default ([GH-93727](https://github.com/godotengine/godot/pull/93727)). This makes
    importing meshes with LODs approximately 2X faster. We expect that the quality of imported LODs will be the same.
    If you notice any issues, please let us know. You can re-enable the raycaster in the object's import settings under
    `lods/raycast_normals`.
- [Editor] Optimize editor grid recalculation ([GH-92734](https://github.com/godotengine/godot/pull/92734)).
- [Editor] Optimize closing the editor settings ([GH-95704](https://github.com/godotengine/godot/pull/95704)).
- [Rendering] Optimize meshes for vertex cache ([GH-94241](https://github.com/godotengine/godot/pull/94241)). In scenes

>     with a lot of geometry this can result in a 20% improvement in frame time. 

Reviewer: ```suggestion
    with a lot of geometry, this can result in a 20% improvement in frame time. 
```

- [Rendering] Optimize materials with `ambient_light_disabled` ([GH-92213](https://github.com/godotengine/godot/pull/92213)).

### And more!

There are too many exciting changes to list them all here, but here are a few more. 

- [Editor] Fix `uid://` paths fail to load at editor startup ([GH-95689](https://github.com/godotengine/godot/pull/95689)).
- [Editor] Correct rotation gizmo plane math for off-center perspective view ([GH-95402](https://github.com/godotengine/godot/pull/95402)).
- [Editor] Make viewport grid visible on all three planes in ortho camera view ([GH-93869](https://github.com/godotengine/godot/pull/93869)).
- [Editor] Expose GDScript syntax highlighter to editor plugins ([GH-95764](https://github.com/godotengine/godot/pull/95764)).
- [Rendering] Fix generated light probes placing too close to manual light probes ([GH-83497](https://github.com/godotengine/godot/pull/83497)).

> - [Rendering] Antialias direct light samples in LightmapperRD ([GH-95828](https://github.com/godotengine/godot/pull/95828)).

Reviewer: I feel this one is worth mentioning in the bicubic sampling paragraph as well, given how they tend to work together to provide the best results.

Reviewer 2: Added.
